tests/unit/c-unit-tests-clang/task.yaml

```diff
@@ -0,0 +1,38 @@
+summary: Build the test suite for C code using clang and run it
+prepare: |
+    # Sanity check, the core snap is installed
+    snap info core | MATCH "installed:"
+    # Install build dependencies for the test
+    dpkg --get-selections > pkg-list
+    # Remove any autogarbage from sent by developer
+    rm -rf "$SPREAD_PATH/cmd/"{autom4te.cache,configure,test-driver,config.status,config.guess,config.sub,config.h.in,compile,install-sh,depcomp,build,missing,aclocal.m4,Makefile,Makefile.in}
+    make -C "$SPREAD_PATH/cmd" distclean || true
+    # install clang
+    . "$TESTSLIB/pkgdb.sh"
+    distro_install_package clang
+execute: |
+    # Refresh autotools build system
+    cd "$SPREAD_PATH/cmd/"
+    autoreconf --install --force
```

Comment:
Maybe we could just run autogen.sh, this would handle the case of what arguments to pass?

Reply:
Ack.